airflow/models.py

```diff
@@ -1594,22 +1656,28 @@ def signal_handler(signum, frame):
                 Stats.gauge(
                     'ti_status',
                     value=0,
-                    tags=[
+                tags=[
                         'task_id:%s' % self.task_id,
                         'dag_id:%s' % self.dag_id,
                         'operator:%s' % self.task.__class__.__name__,
                     ],
                 )
             self.refresh_from_db(lock_for_update=True)
+            self.hostname = None
```

Comment:
Is this to ensure that any tasks pulled think they are on a new host and are released?

Reply:
This is at the end of `TaskInstance._run_raw_task()`, so the objective here is to unset host before the task completes successfully.

In this happy-path case where the task succeeded, this doesn't really do anything - if this hostname unset goes through, the task state change to succeeded will go through, which will prevent the scheduler bounce recovery scenario from occurring.  Still, I think it's nice to be able to query the DB for task instances that have hostname set and be able to say that all of the results are still pending completion.